x/incentive/keeper/keeper.go

```diff
@@ -39,10 +45,267 @@ func NewKeeper(
 		storeKey:   storeKey,
 		memKey:     memKey,
 		paramstore: ps,
-		commitmentKeeper: ck,
+		cmk:        ck,
+		stk:        sk,
+		tci:        &types.TotalCommitmentInfo{},
 	}
 }
 
 func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
 }
+
+// Update total commitment info
+func (k Keeper) UpdateTotalCommitmentInfo(ctx sdk.Context, epochIdentifier string) {
+	// Fetch total staked Elys amount again
+	k.tci.TotalElysBonded = k.stk.TotalBondedTokens(ctx)
+	// Initialize with amount zero
+	k.tci.TotalEdenCommitted = sdk.ZeroInt()
+	k.tci.TotalEdenBoostCommitted = sdk.ZeroInt()
+
+	// Iterate to calculate total Eden and Eden boost committed
+	k.cmk.IterateCommitments(ctx, epochIdentifier, func(commitments ctypes.Commitments) bool {
+		committedTokens := commitments.CommittedTokens
+		// Sum Eden and Eden boost committed
+		for _, c := range committedTokens {
+			// Eden
+			if c.Denom == types.Eden {
+				k.tci.TotalEdenCommitted = k.tci.TotalEdenCommitted.Add(c.Amount)
+			}
+
+			// Eden boost
+			if c.Denom == types.EdenB {
+				k.tci.TotalEdenBoostCommitted = k.tci.TotalEdenCommitted.Add(c.Amount)
+			}
+		}
+
+		return false
+	})
+}
+
+// Calculate total share of staking
+func (k Keeper) CalculateTotalShareOfStaking(amount sdk.Int) sdk.Dec {
+	// Total statked = Elys staked + Eden Committed + Eden boost Committed
+	totalStaked := k.tci.TotalElysBonded.Add(k.tci.TotalEdenCommitted).Add(k.tci.TotalEdenBoostCommitted)
+
+	// Share = Amount / Total Staked
+	return sdk.NewDecFromInt(amount).QuoInt(totalStaked)
+}
```

Comment:
Looks correct to me, we can double check with dream

Reply:
Okay